.github/workflows/buildIOS.yml

```
@@ -0,0 +1,155 @@
+        description: What type of build do you want to run?
+        required: true
+        type: choice
+        options:
+          - release
+          - adhoc
+      ref:
+        description: Git ref to checkout and build
+        required: true
+        type: string
+      pull_request_number:
+        description: The pull request number associated with this build, if relevant.
+        type: number
+        required: false
+
+jobs:
+  build:
+    name: Build iOS app
+    runs-on: macos-13-xlarge
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_15.2.0.app/Contents/Developer
+    outputs:
+      IPA_FILE_NAME: ${{ steps.build.outputs.IPA_FILE_NAME }}
+      DSYM_FILE_NAME: ${{ steps.build.outputs.DSYM_FILE_NAME }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+
+      - name: Create .env.adhoc file based on staging and add PULL_REQUEST_NUMBER env to it
+        if: ${{ inputs.type == 'adhoc' }}
+        run: |
+          cp .env.staging .env.adhoc
+          sed -i '' 's/ENVIRONMENT=staging/ENVIRONMENT=adhoc/' .env.adhoc
+          echo "PULL_REQUEST_NUMBER=${{ inputs.pull_request_number }}" >> .env.adhoc
+
+      - name: Configure MapBox SDK
+        run: ./scripts/setup-mapbox-sdk.sh ${{ secrets.MAPBOX_SDK_DOWNLOAD_TOKEN }}
+
+      - name: Setup Node
+        id: setup-node
+        uses: ./.github/actions/composite/setupNode
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1.190.0
+        with:
+          bundler-cache: true
+
+      - name: Cache Pod dependencies
+        uses: actions/cache@v4
+        id: pods-cache
+        with:
+          path: ios/Pods
+          key: ${{ runner.os }}-pods-cache-${{ hashFiles('ios/Podfile.lock', 'firebase.json') }}
+
+      - name: Compare Podfile.lock and Manifest.lock
+        id: compare-podfile-and-manifest
+        run: echo "IS_PODFILE_SAME_AS_MANIFEST=${{ hashFiles('ios/Podfile.lock') == hashFiles('ios/Pods/Manifest.lock') }}" >> "$GITHUB_OUTPUT"
+
+      - name: Install cocoapods
+        uses: nick-fields/retry@3f757583fb1b1f940bc8ef4bf4734c8dc02a5847
+        if: steps.pods-cache.outputs.cache-hit != 'true' || steps.compare-podfile-and-manifest.outputs.IS_PODFILE_SAME_AS_MANIFEST != 'true' || steps.setup-node.outputs.cache-hit != 'true'
+        with:
+          timeout_minutes: 10
+          max_attempts: 5
+          command: scripts/pod-install.sh
+
+      - name: Decrypt provisioning profiles
+        run: |
+          cd ios
+          provisioningProfile=''
+          if [ '${{ inputs.type }}' == 'release' ]; then
+            provisioningProfile='NewApp_AppStore'
+          else
+            provisioningProfile='NewApp_AdHoc'
+          fi
+          echo "Using provisioning profile: $provisioningProfile"
+          gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output "$provisioningProfile.mobileprovision" "$provisioningProfile.mobileprovision.gpg"
```

Should it have `${ }` like below?

```suggestion
          gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output "${provisioningProfile}.mobileprovision" "${provisioningProfile}.mobileprovision.gpg"
```

---

It's not necessary - the `${ }` is only necessary because `_` is a potentially valid character in a bash variable. So when bash tries to expand the variable, it looks for a variable called `provisioningProfile_Notification_Service`, which does not exist. That's why we have the explicit brackets there.

Definitely a kind of quirky bash thing, but to be consistent about always using `${ }` to expand variables in strings would mean changes in a lot of places.

---

This one def tripped me up